apps/dashboard/src/app/retrieval-agent/agent-dashboard/workflow/workflow.service.ts

```diff
@@ -121,6 +138,85 @@ export class WorkflowService {
   sideBarOpen = computed(() => this._sideBarOpen());
   activeSideBar = computed(() => this._activeSideBar());
 
+  /**
+   * Initialise the workflow with the agents saved in current Arag, update the workflow when current Arag changes.
+   */
+  initAndUpdateWorkflow(root: WorkflowRoot) {
```

Comment:
I guess it is under refactoring and not entirely migrated to the state approach.
The objective is to have a clean separation here:
- when reading the current arag data from the backend, we should just update the state
- when the state changes:
  - some `effect` must create the UI components
  - some `effect` must make the backend calls if needed and if possible (for example, not if a THEN is missing or things like that)